alphastats/llm/llm_integration.py

```diff
@@ -151,8 +151,9 @@ def _truncate_conversation_history(self, max_tokens: int = 100000):
             The maximum number of tokens to keep in history, by default 100000
         """
         # TODO: avoid important messages being removed (e.g. facts about genes)
+        # TODO: find out how messages can be None type and handle them earlier
```

Comment:
this did happen? maybe add some debug code temporarily that prints out the context (probably to `_parse_model_response`:
There we have the whole `response` object. (what we call "message" is actually `response.choices[0].message.content`

Reply:
yes, it happened. I will try again. Could be that this was 8B specific

Reply:
I left this as a todo.